### ISSUE TITLE
Excessive memory for hfst-compose-intersect

### ISSUE THREAD

**Issue by TinoDidriksen** [apertium/apertium-ckt#1]
These steps reached 18 GB RAM each, before I gave up and killed the build:
`hfst-compose-intersect -1 .deps/ckt.LR.lexc-twoc.hfst -2 .deps/ckt.twol.hfst -o .deps/ckt.LR.seg.hfst`
`hfst-compose-intersect -1 .deps/ckt.RL.lexc-twoc.hfst -2 .deps/ckt.twol.hfst -o .deps/ckt.RL.seg.hfst`

**Comment by ftyers** [comment]
Yeah, we basically need to merge in Vasya's code from https://github.com/BasilisAndr/chkchn
